File: swift/DesignPattern.playground/Pages/DesignPrinciple.xcplaygroundpage/Sources/SimplePancakeFactory.swift
import Foundation

public class SimplePancakeFactory {
    public class func createPancake(type: String) -> Pancake? {
        var pancake: Pancake?
        
        // Code that is varying
        switch type {
        case "classic":
            pancake = ClassicPancake()
        case "blueberry":
            pancake = BlueberryPancake()
        case "banana":
            pancake = BananaPancake()
        //    case "chocolate chip":
        //        pancake = ChocolateChipPancake()
        default:
            pancake = ClassicPancake()
        }
        
        return pancake
    }
}

File: kotlin/DesignPattern/src/main/kotlin/designprinciple/source/SimplePancakeFactory.kt
package designprinciple.source

object SimplePancakeFactory {
    fun createPancake(type: String): Pancake {
        return when (type) {
            "classic" -> ClassicPancake()
            "blueberry" -> BlueberryPancake()
            "banana" -> BananaPancake()
//        "chocolate chip" -> ChocolateChipPancake()
            else -> ClassicPancake()
        }
    }
}